### PR TITLE
[DOCS] Splits the roles API documentation into multiple pages

### PR DESCRIPTION
**Pull request by lcawl**
The Role Management API documentation (https://www.elastic.co/guide/en/elasticsearch/reference/master/security-api-roles.html) currently contains information about multiple APIs (get, delete, put, etc). This PR splits that information across multiple pages, mirroring the layout of the machine learning APIs and openAPI format.

This change was discussed briefly in https://github.com/elastic/elasticsearch/pull/32635#discussion_r209154427

### CODE REVIEW

**Comment by lcawl** [comment]
This change can be previewed in https://doc-review-63bf6.firebaseapp.com/es/security-api.html

**Comment by lcawl** [comment]
retest this please

File: x-pack/docs/en/rest-api/security/roles.asciidoc [hunk 1]
```diff
@@ -1,205 +1,9 @@
...
-The Clear Roles Cache API evicts roles from the native role cache. To clear the
-cache for a role, submit a POST request `/_xpack/security/role/<rolename>/_clear_cache`
-endpoint:
-
-[source,js]
---------------------------------------------------
-POST /_xpack/security/role/my_admin_role/_clear_cache
---------------------------------------------------
-// CONSOLE
+* <<security-api-put-role,Create role>>, <<security-api-delete-role,Delete role>>
```
Comment (jaymode) [inline x-pack/docs/en/rest-api/security/roles.asciidoc]: Should delete be on its own line/bullet?

**Review by jaymode — APPROVED** [review]
I left one question, otherwise LGTM

File: x-pack/docs/en/rest-api/security/roles.asciidoc (hunk 1 above)
Comment (lcawl) [reply x-pack/docs/en/rest-api/security/roles.asciidoc]: I've been generally pairing them, but we can re-style if necessary once all the changes are in.

**Pull request merged**